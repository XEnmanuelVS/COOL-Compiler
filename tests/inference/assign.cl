class Main {
    base: AUTO_TYPE <- 1 + 1;
    
    f1(): AUTO_TYPE {
        f2()
    };
    
    f2(): AUTO_TYPE {
        f3()
    };
    
    f3(): AUTO_TYPE {
        f4()
    };
    
    f4(): AUTO_TYPE {
        base
    };
};

