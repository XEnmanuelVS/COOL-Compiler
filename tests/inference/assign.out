\__ProgramNode [<class> ... <class>]
  \__ClassDeclarationNode: class Main  { <feature> ... <feature> }
    \__AttrDeclarationNode: base : Int
    \__FuncDeclarationNode: f1() : Int -> <body>
      \__CallNode: <obj>.f2(<expr>, ..., <expr>)
        \__VariableNode: self

    \__FuncDeclarationNode: f2() : Int -> <body>
      \__CallNode: <obj>.f3(<expr>, ..., <expr>)
        \__VariableNode: self

    \__FuncDeclarationNode: f3() : Int -> <body>
      \__CallNode: <obj>.f4(<expr>, ..., <expr>)
        \__VariableNode: self

    \__FuncDeclarationNode: f4() : Int -> <body>
      \__VariableNode: base

